src/components/FeatureStyler/FeatureStyler.js

```diff
@@ -348,7 +426,36 @@ class FeatureStyler extends PureComponent {
         });
       }
     }
-    return Promise.resolve(newStyle);
+
+    const styleArray = [newStyle, ...extraStyles];
+
+    return Promise.resolve(styleArray);
+  }
+
+  static findLineIcon(styles, feature, lineIcons, start = true) {
+    const geom = feature.getGeometry();
+    for (let i = 1; i < styles.length; i += 1) {
+      if (styles[i].getImage() && styles[i].getImage().getSrc()) {
+        for (let j = 0; j < lineIcons.length; j += 1) {
+          if (styles[i].getImage().getSrc() === lineIcons[j].icon) {
+            const coord = styles[i]
+              .getGeometry()(feature)
+              .getCoordinates();
+            const startCoord = FeatureStyler.getVertexCoord(geom);
+            const endCoord = FeatureStyler.getVertexCoord(geom, false);
+            if (
+              (coord[0] === startCoord[0] &&
+                coord[1] === startCoord[1] &&
+                start) ||
+              (coord[0] === endCoord[0] && coord[1] === endCoord[1] && !start)
```

Comment:
I don't think we need to care about the coordinates, here, we just want to find which icon is used.

Reply:
of course we did, forget it